### ISSUE TITLE
deno remove/uninstall subcommand

### ISSUE THREAD

**Issue by mhvsa** [denoland/deno#3139]
We've got `deno install` and it would be convenient to also have `deno remove` or `deno uninstall` subcommand

**Comment by bartlomieju** [comment]
We had `uninstall` but removed it
https://github.com/denoland/deno_std/pull/512#issuecomment-504005054

**Comment by mhvsa** [comment]
oh, thank you

for myself `rm $(which foo)` works but I think it made sense to keep the `uninstall`

closing

**Comment by roc** [comment]
Just landed in deno land and I was absolutely baffled there's no `uninstall` or `rm` command! Is the reason that you expect a single location for something to be installed, a single version as well? I find it confusing to have an API that has a way of adding something but not of removing it.

**Comment by willgm** [comment]
@mhvsa
> for myself `rm $(which foo)` works but I think it made sense to keep the `uninstall`

What about Window users? Much of the Deno users may not know enough of his system terminal to remove it and feel lost about it.

I believe Deno should have a command to undo something that his own command have made.

**Comment by Yamboy1** [comment]
Even as a linux user I would prefer a specific command to do this rather than having to mess around with shell expansions and the such

**Comment by nateshmbhat** [comment]
> oh, thank you
> 
> for myself `rm $(which foo)` works but I think it made sense to keep the `uninstall`
> 
> closing

The issue with `rm $(which foo)` is that , people can have other binaries than the ones from `deno` that have the same name as deno binaries. This can happen when the `DENO` path is suffixed to the system PATH instead of "prefixing".

So we need a deno uninstall command that will only look in the DENO path and remove the required files

**Comment by llluuulll** [comment]
hi,
so what command i can use to uninstall Deno  ? i am on wondows 10 and i am not at all confortable with complet unknow commands, i dont understand.

run$(witch foo), what is it wich foo ?
can some one point me the correct typo command to use ?

thx.

**Comment by ry** [comment]
> The issue with rm $(which foo) is that , people can have other binaries than the ones from deno that have the same name as deno binaries. This can happen when the DENO path is suffixed to the system PATH instead of "prefixing".

Thanks - good argument. 

I'm ok with something like this:
```
deno install --delete foo
```

**Comment by stefnotch** [comment]
I'd absolutely prefer `deno uninstall` over adding a delete flag to an install command. The latter seems like a odd hack and is going to cause a lot of googling and confusion for newcomers.

**Comment by lem0nify** [comment]
Return `uninstall` subcommand!

`deno uninstall foo` is much more elegant than navigating to and removing the script by hand!

**Comment by trivikr** [comment]
I agree that `uninstall` is a better option, as it's opposite of `install` - similar to `npm install` and `npm uninstall`

The command `remove` is opposite of `add` - similar to `yarn add` and `yarn remove`
The command `delete` is antonym of `insert` - similar to insertion and deletion in data structures.

`deno uninstall` should be preferred opposite for `deno install`

**Comment by whatevergeek** [comment]
i guess while this is still on discussion, we'll have to manage this manually by deciding which ones to delete under the deno folder.
e.g. on Windows 10, C:\Users\<user>\AppData\Local\deno

**Comment by lem0nify** [comment]
@whatevergeek `C:/Users/USERNAME/.deno/bin`, IIRC.
But it's still not the most comfortable place to navigate.

Of course, `uninstall` subcommand is mostly needed in Windows, because Windows users are used to navigate and manage files with Explorer. By default, there is not even a shortcut to the home folder in Windows 10 Explorer and files/folders starting with dot are hidden. Of course, I know I can create my own shortcut and change file hiding policy, but it's weird and unnaturally that deno forces me do it instead of providing a way to uninstall with one simple command what was installed with one simple command.

To be honest, I do not see any objective reason to remove the `uninstall` subcommand, even if it seemed to someone unnecessary.  How could its presence interfere to someone?

**Comment by llluuulll** [comment]
i am totally agee with @lem0nify 
his comment resume perfectly the point

**Comment by wyrd-code** [comment]
I agree, this is an issue for me as well.

Take this case for example, I previously installed `denon` to use local development with a watcher. Now that Deno supports `--watch`, I upgraded and `denon` stopped working. Then I tried to upgrade the `denon` script but I don't see any commands to do that anywhere. After I realized I need to remove it manually (sic), I did a `rm -rf ~/.deno/bin/denon`. Trying to install it again now fails and I seem to be stuck with a message "error: No such file or directory (os error 2)".

If you remove a command like uninstall, the least you can do is refer to it in documentation, explaining why it was removed and how to handle the use case manually.

**Comment by David-Else** [comment]
This seems crazy! Where is the uninstall command?

**Comment by ChoqueCastroLD** [comment]
An uninstall command is the most intuitive thing to do instead of a flag

It will help a lot of newcomers to deno and people who are not so familiarized with their own terminal (windows users mostly)

**Comment by picode7** [comment]
Solution when running on Windows:

```powershell
rm $HOME/.deno/bin/myscript.cmd
```
https://stackoverflow.com/a/62259110/4339170

Please add an uninstall command that works for any system and user command-line knowledge.

**Comment by michaeldesu** [comment]
FWIW an uninstall process for Windows would be appreciated, since I'm on a PC where I need admin-password to do simple things like allow firewall updates. Installing deno was no problem (I had to cancel the admin-controlled firewall/policy update requested by deno post-install) but when I tried to do a upgrade to 1.7, I was blocked by the firewall, so I need to uninstall & reinstall.

**Comment by crowlKats** [comment]
@michaeldesu this feature request is about having a uninstall command for scripts installed through `deno install`, not a way to uninstall deno itself

**Comment by michaeldesu** [comment]
apologies for that

**Comment by einarpersson** [comment]
Having to google `deno uninstall` and coming here is a unnecessary hurdle for a new user.

- If you provide `install` an `uninstall` is expected.
- If the naming had been `add` then `remove` had been expected. 

It is great that the uninstall is really simple under the hood (`rm $(which foo)`) but that does not change the point(s) above at all.

Please reconsider

**Comment by stefnotch** [comment]
Could we please not mark highly upvoted issues as stale?

**Comment by llluuulll** [comment]
Totally agree with einarpersson
Do Deno will consider having a uninstall ?

And also same as stefnotch
Why close an issue if we dont know if is resolve or not

**Comment by merlangoose** [comment]
Here from google, big +1

**Comment by UltiRequiem** [comment]
PLEASE!

**Comment by jcaru** [comment]
I would just like to point out the relevancy of this feature. I personally can't recommend anyone install any software I write for Deno via the install subcommand because the feature is incomplete in that it would require I provide multiple, OS-specific uninstallation commands, rather than the intuitive uninstall subcommand. This looks clunky at best, and unprofessional at worst.

I absolutely love Deno, but I can't make the leap to switch entirely over knowing that an uninstall subcommand is too much to ask. Just as an uninstall subcommand is unnecessary, computers too are unnecessary. Technology is to improve overall productivity, not to hinder it. The lack of an uninstall subcommand because it is unnecessary is akin to opting to write a simple, headless program in assembly because the cross-platform C standard library isn't mandatory to achieve the task.

Deno has achieved so much that Node lacked. Deno may be small, but the software we write with it isn't always as sleek (do >100 MB Electron apps sound familiar?), and what better way to uninstall the app than the intuitive compliment to the graceful `deno install`, `deno uninstall`?

Please reconsider, and thank you for bringing Deno into our dev lives :)

**Comment by crowlKats** [comment]
> uninstall subcommand is too much to ask

It isnt "too much to ask", just not a priority.

**Comment by jcaru** [comment]
Ah, thanks for clarifying @crowlKats. Editing now :)

**Comment by bartlomieju** [comment]
We'll be happy to consider a PR that adds this feature. 

It worth pointing out that it should fail gracefully if a Deno script was not installed in the default location, and suggest appropriate action in that case.

**Comment by dcj9** [comment]
sill nothin eh?

**Comment by dsherret** [comment]
@dcj9 it was released in 1.15.

**Comment by ritamgit-alt** [comment]
How uninstall deno in mac?
any command or something?

**Comment by stefnotch** [comment]
`deno uninstall some-script-name` works with modern versions of Deno

**Comment by ritamgit-alt** [comment]
nope not working with mine

**Comment by dsherret** [comment]
This issue is not about uninstalling Deno. It's about uninstalling commands installed with `deno install`

**Comment by ritamgit-alt** [comment]
so what should I do now?
